File: frontend/src/features/transactions/types/transaction.ts
export interface CreateTransactionPayload {
	paymentIntentId: string;
	senderAccountNumber: number;
	receiverAccountNumber: number;
	securityPin: string;
}

export interface TransactionResponse {
	id: number;
	paymentIntentId: string;
	senderAccountNumber: number;
	senderBankName: string;
	receiverAccountNumber: number;
	receiverOwnerName: string;
	receiverBankName: string;
	status: TransactionStatusType;
	failureReason: string;
	amountTransferred: string;
	timestamp: Date;
	returnUrl?: string;
}

export const TransactionStatus = {
	Successful: "Successful",
	Failure: "Failure",
} as const;

export type TransactionStatusType = (typeof TransactionStatus)[keyof typeof TransactionStatus];

File: python-backend/app/schemas/transaction.py
from datetime import datetime
from decimal import Decimal

from pydantic import Field

from app.schemas.base import CamelModel, CamelResponseModel


class TransactionCreate(CamelModel):
	payment_intent_id: str
	sender_account_number: int
	receiver_account_number: int
	security_pin: str


class TransactionResponse(CamelResponseModel):
	id: int
	payment_intent_id: str
	sender_account_number: int
	sender_owner_name: str
	sender_bank_name: str

	receiver_account_number: int
	receiver_owner_name: str
	receiver_bank_name: str
	status: str
	failure_reason: str | None

	amount_transferred: Decimal = Field(..., examples=["50.00"])
	timestamp: datetime
	return_url: str | None
